### PR TITLE
Upgrade Python 3.11

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: .github/workflows/main-publish.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
```
Comment (netsettler) [inline .github/workflows/main-publish.yml]: 22.04?

File: .github/workflows/main-publish.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install Python dependencies for publish
+        run: python -m pip install dcicutils==7.5.1
```
Comment (netsettler) [inline .github/workflows/main-publish.yml]: This pinning is awkward. Is it right? Is there a way to centralize this better? Some reason `pip install dciciutils` without a version wouldn't work? I assume this same problem occurs in other repos.

File: pyproject.toml [hunk 3]
```diff
@@ -21,12 +21,12 @@ packages = [
 
 
 [tool.poetry.dependencies]
-python = ">=3.7,<3.9"
-dcicutils = "^6.3.0"
-boto3 = "^1.20.15"
+python = ">=3.7,<3.12"
+dcicutils = "7.12.0.1b1"
+boto3 = "^1.28.51"
```
Comment (netsettler) [inline pyproject.toml]: maybe add corresponding boto3-stubs?

**Review by netsettler — APPROVED** [review]
.github/workflow/main.yml needs to be edited to make its test matrix match the approved python versions.

My other comments are not blocking.

File: pyproject.toml [hunk 4]
```diff
@@ -21,12 +21,12 @@ packages = [
 
 
 [tool.poetry.dependencies]
-python = ">=3.7,<3.9"
-dcicutils = "^6.3.0"
-boto3 = "^1.20.15"
+python = ">=3.7,<3.12"
```
Comment (netsettler) [inline pyproject.toml]: Aren't we dropping 3.7?

File: .github/workflows/main-publish.yml (hunk 2 above)
Comment (willronchetti) [reply .github/workflows/main-publish.yml]: Make target that can be reproduced across all repos? I think the concern is somehow bringing in an earlier version that doesn't have the publish command.

**Review by willronchetti — APPROVED** [review]

**Pull request merged**